server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -575,10 +584,7 @@ public synchronized void close() throws IOException {
      */
     public void prepareForClose() {
         HttpServerTransport httpServerTransport = injector.getInstance(HttpServerTransport.class);
-        FutureTask<Void> stopper = new FutureTask<>(() -> {
-            httpServerTransport.stop();
-            return null;
-        });
+        FutureTask<Void> stopper = new FutureTask<>(httpServerTransport::close, null);
```

Comment:
I don't think close will work. AbstractHttpServerTransport shuts down its http channels in stop. Close is a no-op.

Reply:
But `AbstractLifecycleComponent.close` ensures `stop` is called if it is started, _then_ it closes it. This is what we want, no?

Reply:
You are right, I was only looking at close in AbstractHttpServerTransport.